pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -34,6 +34,9 @@ type Constraints struct {
 	Taints Taints `json:"taints,omitempty"`
 	// Requirements are layered with Labels and applied to every node.
 	Requirements Requirements `json:"requirements,omitempty"`
+	// KubeletArgs define args to be used when configuring kubelet on provisioned nodes
+	//+optional
+	KubeletArgs KubeletArgs `json:"kubeletArgs,omitempty"`
```

Comment:
I might just call this `kubelet`, since args is implicit to the fact that you're configuring it. `spec.kubelet.clusterDnsIP` reads quite cleanly.

Reply:
As a consumer of the API spec, I feel the name `spec.kubelet` hints there might be other config beyond just the args. So, that would make me think I'd see something like `spec.kubelet.args`.

Reply:
I guess if we want to make this explicit, should we make it `kubeletConfiguration`: https://github.com/kubernetes/kubelet/blob/master/config/v1beta1/types.go#L88

Reply:
I like that. Just pushed it 👍